src/app/content/highlights/components/EditCard.tsx

```diff
@@ -45,192 +49,415 @@ export interface EditCardProps {
 
 // tslint:disable-next-line:variable-name
 const EditCard = React.forwardRef<HTMLElement, EditCardProps>((props, ref) => {
+  if (!props.isActive) {
+    return null;
+  }
+  return <LoginOrEdit props={props} ref={ref} />;
+});
+
+function LoginOrEdit({
+  props,
+  ref,
+}: {
+  props: React.PropsWithChildren<EditCardProps>;
+  ref: React.ForwardedRef<HTMLElement>;
+}) {
   const authenticated = !!useSelector(selectAuth.user);
-  const loginLink = useSelector(selectAuth.loginLink);
-  const dispatch = useDispatch();
-  const defaultAnnotation = () => props.data && props.data.annotation ? props.data.annotation : '';
-  const [pendingAnnotation, setPendingAnnotation] = React.useState<string>(defaultAnnotation());
-  const [editingAnnotation, setEditing] = React.useState<boolean>(!!props.data && !!props.data.annotation);
-  const [confirmingDelete, setConfirmingDelete] = React.useState<boolean>(false);
   const element = React.useRef<HTMLElement>(null);
-  const textarea = React.useRef<HTMLTextAreaElement>(null);
 
-  const trackEditNoteColor = useAnalyticsEvent('editNoteColor');
-  const trackEditAnnotation = useAnalyticsEvent('editAnnotation');
-  const trackShowCreate = useAnalyticsEvent('showCreate');
-  const trackShowLogin = useAnalyticsEvent('showLogin');
-  const trackDeleteHighlight = useAnalyticsEvent('deleteHighlight');
+  return (
+    <form
+      className={props.className}
+      ref={mergeRefs(ref, element)}
+      data-analytics-region='edit-note'
+      data-highlight-card
+    >
+      {authenticated ? (
+        <ActiveEditCard props={props} element={element} />
+      ) : (
+        <LoginConfirmation data={props.data} onBlur={props.onBlur} />
+      )}
+    </form>
+  );
+}
 
-  const blurIfNotEditing = React.useCallback(() => {
-    if (!props.hasUnsavedHighlight && !editingAnnotation) {
-      props.onBlur();
+function LoginConfirmation({
+  data,
+  onBlur,
+}: Pick<EditCardProps, 'data' | 'onBlur'>) {
+  const loginLink = useSelector(selectAuth.loginLink);
+  const trackShowLogin = useCallbackFor(useAnalyticsEvent('showLogin'));
```

Comment:
instead of wrapping this in a ref utility, if `useAnalyticsEvent` is returning a new function ref every time you call it we should probably fix it internally to use useCallback

Reply:
I think this is a good catch. I was not paying attention to what I wrapped there.